### PR TITLE
Add Android and Flask books, check_urls=free-programming-books-ru.md

### PR DESCRIPTION
**Pull request by kadhirash**
## What does this PR do?
- Add resource

### Description 
- Fixes #4225 
  - Add many Android resources; tried to alphabetize but possible errors since I don't know Russian but used online alphabetizer tool
  - Add Flask resource
### Why is this valuable (or not)?
- Adds many resources; cleans PR
### How do we know it's really free?
- Checked many of the links and they seem to be free since hosted on public websites to read
### For book lists, is it a book?
- Yes
## Checklist:
- [x] Read our [contributing guidelines](/CONTRIBUTING.md)
- [x] Search for duplicates.
- [x] Include author(s) and platform where appropriate.
- [x] Put lists  in alphabetical order, correct spacing.
- [x] Add needed indications (PDF, access notes, under construction)

## Followup

- Check the output of Travis-CI for linter errors!

### CODE REVIEW

File: free-programming-books-ru.md [hunk 1]
```diff
@@ -249,8 +250,16 @@
...
+* [цикл статей по подготовке к собеседованию по android](https://clck.ru/hz2zm) - веселина зацепина
+* [ресурсы, о которых должен знать каждый android-разработчик](https://habr.com/ru/post/175825/)
+* [большая подборка ресурсов для изучения android-разработки](https://tproger.ru/digest/master-android-development/)
+* [освой программирование играючи android](http://developer.alexanderklimov.ru/android/) - александр климов
+* [уроки по разработке андроид-приложений](https://www.fandroid.info)
+* [план изучения android-разработки](https://apptractor.ru/learn/plan-izucheniya-android-razrabotki-dlya-nachinayushhih.html)
+* [как работает android](https://habr.com/ru/company/solarsecurity/blog/334796/)
+* [программирование под android](https://metanit.com/java/android) - евгений попов
+* [уроки по android](http://startandroid.ru/ru/uroki/vse-uroki-spiskom.html)
+* [20 open source проектов для android, которые могут научить вас новому](https://clck.ru/rrmp6)
```
Comment (eshellman) [inline free-programming-books-ru.md:262]: link shortener, 404 not found

File: free-programming-books-ru.md [hunk 2]
```diff
@@ -249,8 +250,16 @@
 
 #### Android
 
-* [Программирование под Android](https://metanit.com/java/android) - Евгений Попов
-* [Уроки по Android](http://startandroid.ru/ru/uroki/vse-uroki-spiskom.html)
+* [цикл статей по подготовке к собеседованию по android](https://clck.ru/hz2zm) - веселина зацепина
```
Comment (eshellman) [inline free-programming-books-ru.md:253]: link shortener, 404 not found

File: free-programming-books-ru.md [hunk 3]
```diff
@@ -249,8 +250,16 @@
...
+* [ресурсы, о которых должен знать каждый android-разработчик](https://habr.com/ru/post/175825/)
+* [большая подборка ресурсов для изучения android-разработки](https://tproger.ru/digest/master-android-development/)
```
Comment (eshellman) [inline free-programming-books-ru.md:255]: not a book - it's a list of other resources.

File: free-programming-books-ru.md [hunk 4]
```diff
@@ -249,8 +250,16 @@
 
 #### Android
 
-* [Программирование под Android](https://metanit.com/java/android) - Евгений Попов
-* [Уроки по Android](http://startandroid.ru/ru/uroki/vse-uroki-spiskom.html)
+* [цикл статей по подготовке к собеседованию по android](https://clck.ru/hz2zm) - веселина зацепина
+* [ресурсы, о которых должен знать каждый android-разработчик](https://habr.com/ru/post/175825/)
```
Comment (eshellman) [inline free-programming-books-ru.md:254]: not a book - it's a list of other resources.

File: free-programming-books-ru.md [hunk 5]
```diff
@@ -249,8 +250,16 @@
...
+* [большая подборка ресурсов для изучения android-разработки](https://tproger.ru/digest/master-android-development/)
+* [освой программирование играючи android](http://developer.alexanderklimov.ru/android/) - александр климов
```
Comment (eshellman) [inline free-programming-books-ru.md:256]: a course, not a book

**Comment by eshellman** [comment]
all the links I've looked at here are not appropriate heere, almost as if the OP copied the list without trying the links.

**Comment by kadhirash** [comment]
@eshellman whoops, yeah I should have looked through each one more thoroughly and not just trust the OP that had a PR for all these links!

**Pull request closed without merging**